src/types/SisoTfTypes/SisoZpk.jl

```diff
@@ -109,32 +103,38 @@ function minreal(sys::SisoZpk{T,TR}, eps::Real) where {T, TR}
     SisoZpk{T, TR}(newZ, newP, sys.k)
 end
 
-# FIXME: Perhaps move to some other file with auxilliary functions,
-# the name could also be imporoved. Perhaps this functionality can be found in some other package.
-""" If TR is Complex and T is Real, check that every pole is matched to its conjugate
-    this assumes that the compelx poles are ordered as they are output by the LAPACK
-    routines that return complex-conjugated values, i.e., (x+iy) is followed by (x-iy)"""
-function check_real(r_vec::AbstractVector{<:Complex})
-    k = 1
-    while k <= length(r_vec)
-        if isreal(r_vec[k])
-            # Do nothing
-        else
-            if k == length(r_vec) # Element is complex and there is no more elements to match up with
-                return false
-            elseif conj(r_vec[k]) == r_vec[k+1] # Element is complex and succeeding element is its conjugate
-                k=k+1
-            else
-                return false
+""" If TR is Complex and T is Real, pair up every pole with its conjugate as they are ordered 
+    by the LAPACK routines that return complex-conjugated values, i.e., (x+iy) is followed by (x-iy)"""
+function pairup_conjugates!(x)
+    i = 0
+    while i < length(x)
+        i += 1
+        imag(x[i]) == 0 && continue
+        foundconj = false
+        for j in i+1:length(x)
+            if x[i] == conj(x[j])
+                if imag(x[i]) > imag(x[j]) # Swap i+1 <-> j
+                    tmp = x[i+1]
+                    x[i+1] = x[j]
+                    x[j] = tmp
+                else # Rotate i -> i+1 -> j -> i
+                    tmp = x[i]
+                    x[i] = x[j]
+                    x[j] = x[i+1]
+                    x[i+1] = tmp
+                end
+                foundconj = true
+                break
             end
         end
```

Comment:
```suggestion

        # Attempt to find a matching conjugate to x[i]
        j = findnext(==(conj(x[i])), x, i+1)
        j === nothing && return false

        tmp = x[j]
        x[j] = x[i+1]
        # Make sure that the complex number with positive imaginary part comes first
        if imag(x[i]) > 0
            x[i+1] = tmp
        else
            x[i+1] = x[i]
            x[i] = tmp
        end
```
Perhaps a bit more readable in certain respects. Hopefully it works :)

Reply:
Agreed, way nicer.